### ISSUE TITLE
Support configuration caching

### ISSUE THREAD

**Issue by joshschriever** [jeremymailen/kotlinter-gradle#200]
It would be great if you could support gradle configuration caching. [The documentation](https://docs.gradle.org/current/userguide/configuration_cache.html) is pretty helpful. I'm guessing it should be pretty straightforward - for `LintTask` it's probably just a matter of injecting `ProjectLayout` and using that instead of `project.projectDir`.

Here's an example of the failure I get when running `./gradlew --configuration-cache test_app:lintKotlinMain`

```
1 problem was found reusing the configuration cache.
- Task `:test_app:lintKotlinMain` of type `org.jmailen.gradle.kotlinter.tasks.LintTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/7.0/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
```

**Comment by plastiv** [comment]
Can fix be published?

**Comment by jeremymailen** [comment]
Very sorry for the delay! It's now published as 3.4.5.
Let me know if it works for you as expected.
